File: code/dobot_quad_sdk-main/high_level/python/examples/e8_rotation.py
#!/usr/bin/env python3
"""Example 8: Rotate in place.
Usage: python e8_rotation.py [server_address] [direction] [angle]
  direction: 0=left/CCW (default), 1=right/CW
  angle:     degrees (default 90)
"""
import sys
from dobot_quad import RobotClient


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    robot.enable_safety_ready()
    direction = int(sys.argv[2]) if len(sys.argv) > 2 else 0
    angle = float(sys.argv[3]) if len(sys.argv) > 3 else 90.0
    direction = 1 if direction == 1 else 0

    print(f"Rotation: {'left (CCW)' if direction == 0 else 'right (CW)'} {angle} deg")
    res = robot.rotate(direction, angle)
    res = robot.circle(direction="right", turns=3)
    return 0 if res and res.success else 1


if __name__ == "__main__":
    sys.exit(main())

File: code/dobot_quad_sdk-main/high_level/cpp/e8_rotation.cpp
// Example 8: Rotate in place.
// Usage: ./e8_rotation [server_address] [direction] [angle]
//   direction: 0=left/CCW (default), 1=right/CW
//   angle:     degrees (default 90)
//
// Simplified API:
//   client.rotate("left", 90);   // left 90 degrees
//   client.rotate_left(90);      // same thing
//   client.rotate_right(45);     // right 45 degrees

#include "robot_client.h"

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    robot::enable_safety_ready(client);
    int dir = (argc > 2) ? std::stoi(argv[2]) : 0;
    float angle = (argc > 3) ? std::stof(argv[3]) : 90.0f;
    dir = (dir == 1) ? 1 : 0;

    std::cout << "Rotation: " << (dir == 0 ? "left (CCW)" : "right (CW)") << " " << angle << " deg" << std::endl;
    client.rotate(dir, angle);
    client.circle(dir == 0 ? "left" : "right", 3);

    return 0;
}
